### PR TITLE
[Pushbullet] Added ability to push notes to a channel

### PR DESCRIPTION
**Pull request by jsetton**
Added the ability to push notes to all subscribers of a channel based on the recipient format. Also, added the ability to push notes without specifying a recipient as the Pushbullet API will default to broadcast them to all the user account's devices.

### CODE REVIEW

File: bundles/action/org.openhab.action.pushbullet/README.md [hunk 1]
```diff
@@ -6,7 +6,9 @@ The Pushbullet action allows you to notify iOS, Android & Windows 10 Phone & Des
 
 The following is a valid action call that can be made when the plugin is loaded.
 For specific information on each item, see the [Pushbullet API](https://docs.pushbullet.com/).
+The recipient can either be an email address or a channel tag. If it is not specified or invalid, the note will be broadcast to all of the user account's devices.
```
Comment (hakan42) [inline bundles/action/org.openhab.action.pushbullet/README.md]: Tiny detail, could you please make sure that every sentence is on its own line?

**Comment by hakan42** [comment]
Looks fine to me. Could you perhaps open and clonse this PR so Jenkins rebuilds it?

The javax.measure error could be caused by the recent Units of Measure feature that came in with the ESH core. Failing a successful build, could you rebase this branch against master and see if that fixes it?

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java [hunk 2]
```diff
@@ -126,7 +133,16 @@ private static boolean sendPush(String botName, String recipient, String title,
...
-        push.setEmail(recipient);
+
+        if (recipient != null) {
+            if (isValidEmail(recipient)) {
+                logger.trace("    Recipient is an email address");
+                push.setEmail(recipient);
+            } else if (isValidChannel(recipient)) {
+                logger.trace("    Recipient is a channel tag");
+                push.setChannel(recipient);
+            }
```
Comment (9037568) [inline bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: else  ?

What happens if neither of these is satisfied? Shouldn't this then throw?

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java [hunk 3]
```diff
@@ -126,7 +133,16 @@ private static boolean sendPush(String botName, String recipient, String title,
         push.setTitle(title);
         push.setBody(body);
         push.setType(type);
-        push.setEmail(recipient);
+
+        if (recipient != null) {
```
Comment (9037568) [inline bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: else ?

What if recipient is null? Shouldn't there be a throw in that case?

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java (hunk 2 above)
Comment (jsetton) [reply bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: If neither of these is satisfied, no specific target will be included in the Pushbullet API query.

Per the [API documentation](https://docs.pushbullet.com/#target-parameters), if no target is specified, the notification will be broadcast to all of the user's devices, which is considered as the default behavior.

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java (hunk 3 above)
Comment (jsetton) [reply bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: Same reason as above. If the recipient is null, we would fallback to the default behavior.

**Comment by jsetton** [comment]
I have updated the README file per @hakan42 recommendation and also rebased the branch against master. All checks are now passing.

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java (hunk 2 above)
Comment (jsetton) [reply bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: I added an log warn to let the user know that the provided recipient was invalid.

File: bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java [hunk 4]
```diff
@@ -126,7 +136,19 @@ private static boolean sendPush(String botName, String recipient, String title,
...
+            if (isValidEmail(recipient)) {
+                logger.trace("    Recipient is an email address");
+                push.setEmail(recipient);
+            } else if (isValidChannel(recipient)) {
+                logger.trace("    Recipient is a channel tag");
+                push.setChannel(recipient);
+            } else {
+                logger.warn("Invalid recipient: {}", recipient);
```
Comment (9037568) [inline bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: These two should be DEBUG, not WARN.

Comment (jsetton) [reply bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: Shouldn't we let the user know that he specified an invalid recipient? Sending these two in DEBUG will hide this information.

Comment (9037568) [reply bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java]: I overlooked the distinction between 'not specified' and 'specified, but invalid'.
With the view that this warning is for the latter case, I agree with you.

**Comment by 9037568** [comment]
Thanks, @jsetton !

**Pull request merged**